src/bun.js/web_worker.zig

```diff
@@ -55,17 +50,10 @@ pub const WebWorker = struct {
     pub fn hasPendingActivity(this: *WebWorker) callconv(.C) bool {
         JSC.markBinding(@src());
 
-        if (this.vm == null) {
-            return !this.requested_terminate;
-        }
-
-        if (!this.allowed_to_exit) {
-            return true;
+        if (this.vm) |vm| {
+            return vm.eventLoop().tasks.count > 0 or vm.active_tasks > 0 or vm.uws_event_loop.?.active > 0;
```

Comment:
I think this method might not actually be called

but if it is, this code isn't threadsafe

Reply:
this was just refactoring the left without allowed_to_exit; so the original code wouldn't be threadsafe either?